### PR TITLE
Themes analytics and availability

### LINKED ISSUE #3914: Make the themes feature available to all users
Once the feature is mature enough, we should make it public.  This will involve cleaning up some of the code we added previously in [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/3913).

### LINKED ISSUE #4177: Add tracking to themes.
Add tracking to themes, once the rest of the features are implemented.

Discuss with @sendhil and @folletto to define the events we want / need, and update this ticket.

### PR DESCRIPTION
**Pull request by alexcurylo**
Closes #3914; closes #4177

Adds theme analytics matching Android implementation; opens new theme browser implementation to all users.

To test: Install where themes feature has not been previously enabled with wpdebug://themes?enabled=1 as discussed in #3913 and ensure that themes browser is available in blog details screen.

Needs review: @astralbodies
cc: @kwonye

### CODE REVIEW

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m [hunk 1]
```diff
@@ -272,11 +272,31 @@ - (WPAnalyticsTrackerMixpanelInstructionsForStat *)instructionsForStat:(WPAnalyt
...
+        case WPAnalyticsStatThemesDemoAccessed:
+            instructions = [WPAnalyticsTrackerMixpanelInstructionsForStat mixpanelInstructionsForEventName:@"Themes - Demo Accessed"];
+            break;
+        case WPAnalyticsStatThemesDetailsAccessed:
+            instructions = [WPAnalyticsTrackerMixpanelInstructionsForStat mixpanelInstructionsForEventName:@"Themes - Details Accessed"];
+            break;
+        case WPAnalyticsStatThemesPreviewedSite:
+            instructions = [WPAnalyticsTrackerMixpanelInstructionsForStat mixpanelInstructionsForEventName:@"Themes - Previewed Theme for Site"];
+            [instructions setSuperPropertyAndPeoplePropertyToIncrement:@"number_of_times_previewed_a_theme"];
+            [instructions setCurrentDateForPeopleProperty:@"last_time_previewed_a_theme"];
```
Comment (kwonye) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerMixpanel.m]: Great idea! I'll add this tracking as well to Android

**Comment by kwonye** [comment]
Looks consistent with Android, so all good there :+1:

**Comment by astralbodies** [comment]
Reviewing now, @alexcurylo

**Comment by astralbodies** [comment]
Code looks good! :shipit: 

I filed a bug relating to Themes that isn't part of this PR - #4574

**Comment by astralbodies** [comment]
@daniloercoli I added these Tracks events to the spreadsheet - they pair up with yours perfectly :wink:

**Comment by alexcurylo** [comment]
Thank you very much @astralbodies!

**Pull request merged**